reproject/healpix/utils.py

```diff
@@ -0,0 +1,53 @@
+import tempfile
+
+import numpy as np
+from healpy import read_map
+
+from astropy.io.fits import TableHDU, BinTableHDU
+from astropy.extern import six
+from astropy.coordinates import BaseCoordinateFrame, frame_transform_graph, Galactic, ICRS
+
+FRAMES = {
+    'g': Galactic(),
+    'c': ICRS()
+}
+
+
+def parse_coord_system(system):
+    if isinstance(system, BaseCoordinateFrame):
+        return system
+    elif isinstance(system, six.string_types):
+        system = system.lower()
+        if system == 'e':
+            raise ValueError("Ecliptic coordinate frame not yet supported")
+        elif system in FRAMES:
+            return FRAMES[system]
+        else:
+            system_new = frame_transform_graph.lookup_name(system)()
+            if system_new is None:
+                raise ValueError("Could not determine frame for system={0}".format(system))
+            else:
+                return system_new
+
+
+def parse_input_healpix_data(input_data, field=0, hdu_in=None):
+    """
+    Parse input HEALPIX data to return a Numpy array and coordinate frame object.
+    """
+
+    if isinstance(input_data, (TableHDU, BinTableHDU)):
+        # TODO: for now we have to write out to a temporary file
```

Comment:
What do you mean by "for now"?
Is there a feature request in healpy or Astropy to implement something that's needed here?
Maybe leave a reference to it here or as a `reproject` Github issue?